ci-operator/templates/openshift/installer/cluster-launch-installer-upi-e2e.yaml

```diff
@@ -457,6 +1096,53 @@ objects:
           ${SSH_PUB_KEY}
         EOF
 
+          # Add proxy info to install-config.yaml
+          if [[ "${ENABLE_PROXY}" == "true" ]]; then
+
+            # create and sign a cert
+            generate_proxy_certs
+            ROOTCA=/tmp/CA
+            INTERMEDIATE=${ROOTCA}/INTERMEDIATE
+
+            # load in certs here
+            PROXY_CERT="$(base64 -w0 ${INTERMEDIATE}/certs/intermediate.cert.pem)"
+            PROXY_KEY="$(base64 -w0 ${INTERMEDIATE}/private/intermediate.key.pem)"
+            PROXY_KEY_PASSWORD="$(cat ${ROOTCA}/intpassfile)"
+
+            CA_CHAIN="$(base64 -w0 ${INTERMEDIATE}/certs/ca-chain.cert.pem)"
+
+            # create random uname and pw
+            USER_NAME="${CLUSTER_NAME}"
+            PASSWORD="$(uuidgen | sha256sum | cut -b -32)"
+
+            HTPASSWD_CONTENTS="${USER_NAME}:"$(openssl passwd -apr1 ${PASSWORD})""
+            HTPASSWD_CONTENTS="$(echo -e ${HTPASSWD_CONTENTS} | base64 -w0)"
+
+            KEY_PASSWORD="$(base64 -w0 << EOF
+        #!/bin/sh
+        echo ${PROXY_KEY_PASSWORD}
+        EOF
```

Comment:
What's this about?  Does something like `KEY_PASSWORD="$(cat ${ROOTCA}/intpassfile | base64 -w0)"` not do the trick and allow us to drop `PROXY_KEY_PASSWORD`?

Reply:
this is due to the fact that we have a password on our key (given its AES), this is so squid can have access to the password [1]

[1] https://github.com/openshift/release/pull/4719/files#diff-2b1b845b92f8062711789a2bfdb27290R1342

Reply:
Oh, I see, this is shipping up a script, not just the password itself.  Fine for now, we can always revisit if someone has a cleaner idea later (I don't have one now ;).

Reply:
its the documented method by squid for dealing with certs that require a password